CIV-5274 - Remove handle transaction implementation

leolangen opened the pull request:
The new version of the Civic React Component implements an internal client-sends handleTransaction function, meaning that this complexity can be removed from dApp implementations.

This PR removes ‘handleTransaction’ as a Civic GatewayProvider parameter and instead adds a smaller, cleaner useEffect to use the GatewayStatus going ACTIVE to trigger minting. It also bump the Civic React Component version to the latest, 0.9.2

Also in this PR is a bug fix for a bug where the UI was hanging in some error cases because of an unhandled promise in a call to `connection.removeSignatureListener(subId);`

leolangen committed:
feat: remove handle transaction callback

- create useEffect in mint button to trigger on mint
when status changes to active

leolangen committed:
fix: prevent unhalded promise rejections

- add await to connection.removeSignatureListener
to ensure that a possible promise rejection is handled

leolangen committed:
feat: use latest RC alpha

kevinhcolgan commented on src/Home.tsx:
```diff
@@ -307,6 +306,7 @@ const Home = (props: HomeProps) => {
   ) => {
     try {
       setIsUserMinting(true);
+      console.log(`document.getElementById("#identity"): ${document.getElementById("#identity")}`);
```
should be removed

leolangen replied:
good catch, should I also remove the click below?

kevinhcolgan commented on src/Home.tsx:
```diff
@@ -613,66 +613,6 @@ const Home = (props: HomeProps) => {
-                      }
-                      await onMint();
-                    }}
                     broadcastTransaction={false}
```
we can also remove the broadcastTransaction flag as it's now the default

leolangen replied:
sure, will remove

leolangen committed:
feat: cleanup

- remove unneeded broadcastTransaction
- remove log and identity element click
that doesn't exist

leolangen committed:
feat: use 0.9.2-beta.1

kevinhcolgan committed:
feat: use const for CIVIC_GATEKEEPER_NETWORK

kevinhcolgan committed:
feat: bump civic react component 0.9.2
